### ISSUE TITLE
Barsuk hatch ladder bots unable to get off

### ISSUE THREAD

**Issue by MateuszDlugi** [Regalis11/Barotrauma#10686]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

After departing from an outpost bots started repairing stuff and got stuck on the top part of the ladder on Barsuk. Honestly I have no idea what led to that issue. I was using some mods for previous multi gameplay: "Lua For Barotrauma", "Multiplayer crew manager", "CsForBarotrauma". The single palyer save was generated with those mods but I tested the issue without mods too, on the same save of course as I have no idea how to reproduce the problem. Every time I boot the save same thing happenes. Also I didnt find anything to repair on top floor. Tried taking control of the bots but same thing happens when I switch. One more thing is that it happened twice but first time was during combat so I just handled everything myself and thought the bug wil go away after next departure.
Additionaly I recorded the issue and posted on yt [here](https://youtu.be/9Z59yBruVSE)

### Bug prevalence

Just once

### Version

0.19.14.0

### Which operating system did you encounter this bug on?

Windows

**Comment by ekqvist** [comment]
Fixed with an additional waypoint on the ladder near the top to help the bots get to the hatch and fix it (Regalis11/Barotrauma-development@7a36ce2).

I remember that the platform below the top hatch was added way back when to help bots fix the hatch. I don't know if the changes to the logic when implementing the fixing on ladders somehow caused them to get confused and unable to use the platform to fix the hatch anymore. But they should be doing it on the ladder now and not get stuck on it anymore.

**Comment by itchyOwl** [comment]
Yeah, platforms were a workaround and aren't actually needed anymore.

**Comment by ekqvist** [comment]
We could even consider removing them altogether. It might make some of the navigation a bit smoother now that they're not needed. But I didn't want to jump the gun on that yet. Something to consider.

**Comment by NilanthAnimosus** [comment]
Tested against bugfixes commit https://github.com/Regalis11/Barotrauma-development/commit/0b2d49ccd119c0e932389f71cf6fa7ab6d7680f0.

Bots did not get stuck on the ladder and could repair the junctions/hatch and such. however they were unable to weld either side of the hatch and could not navigate to it.

**Comment by ekqvist** [comment]
Ok, that's weird since I did also test they could weld the walls. But I will test it again and see what's going wrong. Maybe I missed something.

**Comment by ekqvist** [comment]
Based on my testing, the new issue seems to be the following:

The bots will climb the ladder as far as they can to get to the closest waypoint possible. Once they're there, they will perform a check to see if there's something in the way before they start welding the leak. The top hatch will block them from welding as it gets in the way and they won't try to weld the leak in order to avoid also welding shut the hatch.
